target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch

```diff
@@ -63,13 +61,15 @@
 +	dev_info(&pdev->dev, "Revision : %s\n", hw.rev);
 +
 +	if (!strcmp(hw.board, "LINKITS7688")) {
```

Comment:
I suggest removing this patch. We can control this bootstrap pin (GPIO#11) in device tree via gpio-export or gpio-hog. BTW, I don't see this GPIO on Pin-out diagram. It's a hack for custom modified hardware?
![image](https://github.com/user-attachments/assets/792684b7-5c48-43b2-a952-97f28bfedbd7)
![image](https://github.com/user-attachments/assets/8bc1903e-dace-4854-833c-60963d85c45e)

Reply:
Got it.

Reply:
I think you can change the title and make this change here if you like.

Reply:
Something like this?

```
--- a/target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts
+++ b/target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts
@@ -22,10 +22,16 @@
                serial0 = &uart2;
        };
 
-       bootstrap {
-               compatible = "mediatek,linkit";
+       export {
+               compatible = "gpio-export";
+               #size-cells = <0>;
+
+               bootstrap {
+                       gpio-export,name = "bootstrap";
+                       gpio-export,output = <1>;
+                       gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+               };
 
-               status = "okay";
        };
 
        leds {

```

Reply:
Yes. The gpio-export should be fine.

Finally I know, this pin (GPIO#11, aka GPIO 0) is used to control the bus switch chip. Output low level will enable export i2s, spi and uart bus. ref: https://seeeddoc.github.io/LinkIt_Smart_7688/res/Hardware_Schematics.zip
![image](https://github.com/user-attachments/assets/007f27d0-27bf-4d8d-a90d-4a92e6d606c4)

Reply:
Maybe gpio-hog is better. Not sure. I don't know if this really needs to be controlled.